### PR TITLE
[Proto Annotations] Allow for null configProto; small changes

### PR DESCRIPTION
**Pull request by andreamlin**
- Pass in list of ProtoFiles to parse interfaces from

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTransformer.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ private ViewModel buildGapicClientViewModel(GapicInterfaceContext context) {
     apiImplClass.protoFilename(context.getInterface().getFile().getSimpleName());
     String implName = namer.getApiWrapperClassImplName(context.getInterfaceConfig());
     apiImplClass.name(implName);
-    String serviceAddress = context.getModel().getServiceConfig().getName();
+    String serviceAddress = context.getServiceAddress();
```
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTransformer.java]: This change is to follow the convention set by https://github.com/googleapis/gapic-generator/pull/2328

**Comment by andreamlin** [comment]
PTAL

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java [hunk 2]
```diff
@@ -139,31 +140,49 @@ public static GapicProductConfig create(
...
-          .getDiagCollector()
-          .addDiag(
-              Diag.error(
-                  SimpleLocation.TOPLEVEL,
-                  "There are no source proto files with package %s",
-                  defaultPackage));
+
+    if (protoPackage != null) {
+      if (configProto == null) {
+        if (sourceProtos.size() == 0) {
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: Let's use `isEmpty` instead.

File: src/main/java/com/google/api/codegen/config/GapicProductConfig.java [hunk 3]
```diff
@@ -139,31 +140,49 @@ public static GapicProductConfig create(
...
-          .getDiagReporter()
-          .getDiagCollector()
-          .addDiag(
-              Diag.error(
-                  SimpleLocation.TOPLEVEL,
-                  "There are no source proto files with package %s",
-                  defaultPackage));
+
+    if (protoPackage != null) {
+      if (configProto == null) {
```
Comment (pongad) [inline src/main/java/com/google/api/codegen/config/GapicProductConfig.java]: `if (protoPackage != null && configProto == null)`?

**Pull request merged**